CompilerSource/ipc/Room.fbs

```diff
@@ -0,0 +1,113 @@
+
+table RoomInstance {
+  id:int;
+
+  x:int;
+  y:int;
+  object_id:int;
+
+  flip_x:bool;
+  flip_y:bool;
+  scale_x:double;
+  scale_y:double;
+  rotation:double;
+  color:int;
+
+  creation_code:string;
+  locked:bool;
+}
+
+struct RoomTile {
+  id:int;
+
+  background_id:int;
+  bg_x:int;
+  bg_y:int;
+  room_x:int;
+  room_y:int;
+  width:int;
+  height:int;
+  depth:int;
+
+  scale_x:double;
+  scale_y:double;
+  color:int;
+
+  locked:bool;
+}
+
+struct RoomBackground {
+  visible:bool;
+  foreground:bool;
+  background_id:int;
+  x:int;
+  y:int;
+  tile_horizontal:bool;
+  tile_vertical:bool;
+  speed_h:int;
+  speed_v:int;
+  stretch:bool;
+}
+
+struct RoomView {
+  visible:bool;
+  x:int;
+  y:int;
+  width:int;
+  height:int;
+  port_x:int;
+  port_y:int;
+  port_w:int;
+  port_h:int;
+  border_h:int;
+  border_v:int;
+  speed_h:int;
+  speed_v:int;
+  object_id:int;
+}
+
+struct RoomEditorMetadata {
+  remember_window_size:bool;
+  editor_width:int;
+  editor_height:int;
+  current_tab:int;
+  scroll_x:int;
+  scroll_y:int;

```

Comment:
Now, I know we said we want to keep the editor metadata to serialize it. But this is really only important for the other two groups of fields below this which are genuine settings and not plain old metadata.

The editor size itself is something we would honestly want the IDE to remember for **_EVERY_** editor. GMX does serialize this in the GMX. But I'm thinking of something bigger that would serialize the entire state of the IDE for every editor and dock widget when you close the window.

The disadvantage to this is then that it gets serialized to the registry or wherever a settings object in wx or Qt would put it thus making it less portable. The disadvantage is that if you take your project somewhere and open it on another computer, the entire layout is forgotten.

One possibility would be to continue serializing all of this in the project, but put it in a separate metadata file at the root of the project. The reason being is it's junk and **_people don't want that in their git logs_**. This is actually the case with a Qt or CodeBlocks project itself where I added the persistent layout to gitignore.